.github/workflows/go.yml

```diff
@@ -39,6 +39,9 @@ jobs:
       - name: Run the unit tests
         run: make test
 
+      - name: Check for race conditions
```

Comment:
Shouldn't the race detection be default for the default test command?

Reply:
Test without race is faster than with race.
Hence the idea was to fail fast and early in case of a faulty test.

But since there is no need at the moment for this performance optimization I will add it to the test make target and remove the race target

Reply:
fixed